scripts/lsp_types.lua

```diff
@@ -18,6 +19,84 @@ local function tofile(fname, text)
   end
 end
 
+local function gen_mehtods(opt)
+  local uri = 'https://raw.githubusercontent.com/microsoft/language-server-protocol/gh-pages/_specifications/lsp/'
+    .. opt.version
+    .. '/metaModel/metaModel.json'
+
+  local res = vim
+    .system({
+      'curl',
+      uri,
+      '-o',
+      '-',
+    })
+    :wait()
+
+  if res.code ~= 0 then
+    print(res.stderr)
+    return
+  end
+  res = vim.json.decode(res.stdout)
+  local output = { '--Lsp methods keep at last', 'protocol.Methods = {' }
+  local indent = (' '):rep(2)
+  vim.iter(res.requests):map(function(item)
+    if item.method then
+      local name = ''
+      --remove textDoucment prefix but keep workspace prefix
+      --method name convert to CamelCase
```

Comment:
This conversion won't be needed if following @justinmk's [suggestion of using the same casing as the protocol](https://github.com/neovim/neovim/pull/24504#discussion_r1278152406) right?